### PR TITLE
Futures remove closure fee and rounding

### PR DESCRIPTION
**Pull request by artdgn**
### Warning: This PR is on top of `futures-next-price` branch (https://github.com/Synthetixio/synthetix/pull/1609) so should be merged only **AFTER** that PR is merged. But can be reviewed already.

Implements https://github.com/Synthetixio/issues/issues/380

Also removes usage of rounding for decimal operations in futures contracts and tests.

The order fee calculation has been simplified to only consider the direction of the trade w.r.t. the skew before the trade. This makes the code simpler, more efficient, and easier to reason about and test. This means that in the edge-case of a trade that flips the skew - the fee will be charged as if it doesn't. This should be negligible both because this will a rare edge case, and because the difference in that case is very small (whether the full fee is all taker, all maker, or a combination).

### CODE REVIEW

File: contracts/FuturesMarketBase.sol [hunk 1]
```diff
@@ -247,7 +247,7 @@ contract FuturesMarketBase is Owned, Proxyable, MixinFuturesMarketSettings, IFut
      */
     function _maxOrderSizes(uint price) internal view returns (uint, uint) {
         (uint long, uint short) = _marketSizes();
-        int sizeLimit = int(_maxMarketValueUSD(baseAsset)).divideDecimalRound(int(price));
+        int sizeLimit = int(_maxMarketValueUSD(baseAsset)).divideDecimal(int(price));
```
Comment (jacko125) [inline contracts/FuturesMarketBase.sol:250]: This would reduce the `sizeLimit` capacity slightly (doesn't round up).

File: contracts/FuturesMarketBase.sol [hunk 2]
```diff
@@ -648,7 +647,7 @@ contract FuturesMarketBase is Owned, Proxyable, MixinFuturesMarketSettings, IFut
      */
     function _liquidationFee(int positionSize, uint price) internal view returns (uint lFee) {
         // size * price * fee-ratio
-        uint proportionalFee = _abs(positionSize).multiplyDecimalRound(price).multiplyDecimalRound(_liquidationFeeRatio());
+        uint proportionalFee = _abs(positionSize).multiplyDecimal(price).multiplyDecimal(_liquidationFeeRatio());
```
Comment (jacko125) [inline contracts/FuturesMarketBase.sol:650]: Same comment here it'll truncate instead of rounding the multiply / division

**Review by jacko125 — APPROVED** [review]
LGTM, only comment is no rounding in certain calculations for the market limits and liquidation calcs, accessible margin
